dataland-frontend/tests/e2e/specs/sme/DataIntegrity.ts

```diff
@@ -0,0 +1,98 @@
+import { type FixtureData, getPreparedFixture } from "@sharedUtils/Fixtures";
+
+let smeFixtureForTest: FixtureData<SmeData>;
+before(function () {
+  cy.fixture("CompanyInformationWithSmePreparedFixtures").then(function (jsonContent) {
+    const preparedFixturesSme = jsonContent as Array<FixtureData<SmeData>>;
+    smeFixtureForTest = getPreparedFixture("Sme-dataset-with-no-null-fields", preparedFixturesSme);
+  });
+});
+
+describeIf(
+  "As a user, I expect to be able to edit and submit Sme data via the upload form",
+  {
+    executionEnvironments: ["developmentLocal", "ci", "developmentCd"],
+  },
+  function (): void {
+    let tokenForAdminUser: string;
+    let storedTestCompany: StoredCompany;
+    let dataMetaInfoOfTestDataset: DataMetaInformation;
+    before(() => {
+      const uniqueCompanyMarker = Date.now().toString();
+      const testCompanyName = "Company-Created-In-Sme-Blanket-Test-" + uniqueCompanyMarker;
+      getKeycloakToken(admin_name, admin_pw)
+        .then((token: string) => {
+          tokenForAdminUser = token;
+          return uploadCompanyViaApi(token, generateDummyCompanyInformation(testCompanyName));
+        })
+        .then((storedCompany) => {
+          storedTestCompany = storedCompany;
+          return uploadFrameworkData(
+            DataTypeEnum.Sme,
+            tokenForAdminUser,
+            storedCompany.companyId,
+            "2021",
+            smeFixtureForTest.t,
+          );
+        })
+        .then((dataMetaInfo) => {
+          dataMetaInfoOfTestDataset = dataMetaInfo;
+        });
+    });
+
+    it(
+      "Create a company and a Sme dataset via api, then re-upload it with the upload form in Edit mode and " +
+        "assure that the re-uploaded dataset equals the pre-uploaded one",
+      () => {
+        cy.ensureLoggedIn(admin_name, admin_pw);
+        cy.intercept("**/api/companies/" + storedTestCompany.companyId + "/info").as("getCompanyInformation");
+        cy.visitAndCheckAppMount(
+          "/companies/" +
+            storedTestCompany.companyId +
+            "/frameworks/" +
+            DataTypeEnum.Sme +
+            "/upload?templateDataId=" +
+            dataMetaInfoOfTestDataset.dataId,
+        );
+        cy.wait("@getCompanyInformation", { timeout: Cypress.env("medium_timeout_in_ms") as number });
+        cy.get("h1").should("contain", storedTestCompany.companyInformation.companyName);
+        cy.intercept({
+          url: `**/api/data/${DataTypeEnum.Sme}`,
+          times: 1,
+        }).as("postCompanyAssociatedData");
+        submitButton.clickButton();
+        cy.wait("@postCompanyAssociatedData", { timeout: Cypress.env("medium_timeout_in_ms") as number })
+          .then((postInterception) => {
+            cy.url().should("eq", getBaseUrl() + "/datasets");
+            const dataMetaInformationOfReuploadedDataset = postInterception.response?.body as DataMetaInformation;
```

Comment:
Is the datatype of the response body not already companyassociateddata and the request below is therefore uneccessary overhead?

Reply:
Ah ne das ist vom naming ein bisschen irreführend:

Diese `postInterception` ist ja die Antwort vom backend. Und die Antwort ist vom Type `DataMetaInformation`.
Daher passt das.

Du lädst ein sme-dataset hoch, nimmst die DataMetaInfo die zurück kommt, rufst dieses dataset damit vom backend ab, und verlgeichst es mit dem ursprünglich hochzuladendenden fixture-dataset.

Und die zwei sollen gleich sein.

Reply:
Ich nenn das mal um zu `postResponseInterception`

Reply:
Ah, stimmt ja. Dennoch gibt es auch `postInercetpion.request.body` glaube ich, wo die Daten ja direkt drinstecken.

Reply:
Achso jetzt verstehe ich besser was du meinst.

Ja das könnte ich machen, aber ist es nich sinnvoller, den tatsächlich gespeicherten Datensatz nochmal abzurufen und mit dem fixture zu vergleichen?

Wenn ich nur das abgesendete Dataset mit dem fixture vergleiche, dann kriege ich es nicht mit, wenn das backend beim parsen und speichern irgendwas vergeigt.

Außerdem wäre das dann nicht konsistent zu den anderen blanket-tests, wo wir das dataset auch nochmal abrufen, und nicht aus dem request.body nehmen.

Reply:
Echt? Machen wir das überall nochmal mit nem API call. Dann meinetwegen. Aber eigentlich ne sichere Sache, dass das Backend die Daten nicht manipuliert.